ui/editor/src/ctrl.ts

```diff
@@ -133,6 +133,40 @@ export default class EditorCtrl {
     );
   }
 
+  private isLikelyStandard(legalFen: string): boolean {
+    const setup = parseFen(legalFen).unwrap();
+    const board = setup.board;
+
+    if (
+      setup.unmovedRooks
+        .without(0)
+        .without(7)
+        .without(7 * 8)
+        .without(7 * 8 + 7)
+        .nonEmpty()
+    )
+      return false;
+
+    for (const color of COLORS) {
+      const pieces = board[color];
+      const canCastle = setup.unmovedRooks.intersect(SquareSet.backrank(color)).nonEmpty();
+      const expectedKingSquare = color === 'white' ? 4 : 7 * 8 + 4;
+
+      if (canCastle && board.king.intersect(pieces).singleSquare() !== expectedKingSquare) return false;
+
+      const promotedPieces =
+        Math.max(board.queen.intersect(pieces).size() - 2, 0) +
```

Comment:
Shouldn't we subtract by one for the queen (i.e. `board.queen.intersect(pieces).size() - 1`) ?

Reply:
Yes, you're right. Fixed.